### PR TITLE
Add readmes

### PR DESCRIPTION
**Pull request by pet-mit**
- Add CONTRIBUTING guide
- Add code formatting configuration
- Add README for functional tests
- Add more details in doc README
- Fix issues with existing doc

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+3. The [files](src/test/resources/files) directory contains the file resources for the tests (CRACs, Networks, etc) 
+   seperated according to their business type.
+4. The [java source](src/test/java/com/powsybl/openrao/tests) directory contains code that ["glues"](#glue)
+   the [Gerkin features](#features) to the actual OpenRAO code.
+
+### Features
+The [features](src/test/resources/features) directory contains the "feature" files, that contain the actual description
+of the features, written in [Gerkin](https://cucumber.io/docs/gherkin/) syntax.
+
+#### Feature description
```
Comment (phiedw) [inline tests/README.md:27]: this whole paragraph has features and scenarii mixed up

File: tests/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+- A "feature" file actually contains multiple features (tests), grouped into one "scenario"
+- A scenario has a title and a number (OpenRAO was historically developed following the Agile methodology, so epic numbers were used)
+- Every feature has a number and a brief title (user story numbers were used, but we continue to use numbers in order to easily identify tests)
+- Every feature can have a set of tags (`@...`) that allows advanced filtering of features
+
+#### General feature structure
+A feature is a collection of "steps", which is an instruction written in (almost) natural language, starting with `Given`,
```
Comment (phiedw) [inline tests/README.md:34]: Also need to change feature to scenario in this paragraph

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+As you can see, the body of the `iExportCoreCne` method simply calls the CORE CNE API of OpenRAO.  
+
+OpenRAO's glue code is located in the [test](src/test/java/com/powsybl/openrao/tests) directory.
+
+## Contributing
+If your development changes a behavior, you may need to update the tests.  
+- If you add a new behavior to OpenRAO, try to add new "features" that test this behavior, in different contexts
+- If you changed or improved some behavior and broke a test, you have to update it. Whenever possible, instead of only 
+  changing a given test, try to duplicate it in order to test the old behavior (if it is still possible, using a 
+  specific configuration), and the new one, in two different features.
```
Comment (phiedw) [inline tests/README.md:67]: scenarios

**Review by phiedw — CHANGES_REQUESTED** [review]

File: tests/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+- a feature is a collection of related scenarios
+
+## Source code content
+The source code of this package is divided in four main parts:
+1. The [RunCucumberTest](src/test/java/com/powsybl/openrao/tests/RunCucumberTest.java) main class is a test class that 
+   runs the Cucumber tests through a JUnit plugin
+2. The [features](src/test/resources/features) directory contains the "feature" files, that contain the actual description 
+   of the features, written in [Gerkin](https://cucumber.io/docs/gherkin/) syntax (see [Features](#features)).
+3. The [files](src/test/resources/files) directory contains the file resources for the tests (CRACs, Networks, etc) 
+   seperated according to their business type.
```
Comment (OpenSuze) [inline tests/README.md:19]: separated

File: tests/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+```java
+@When("I export CORE CNE at {string}")
+public void iExportCoreCne(String timestamp) throws IOException {
+    cneVersion = CneHelper.CneVersion.CORE;
+    CommonTestData.loadData(timestamp);
+    exportedCne = CneHelper.exportCoreCne(CommonTestData.getCrac(), CommonTestData.getCracCreationContext(), CommonTestData.getNetwork(), CommonTestData.getRaoResult(), CommonTestData.getRaoParameters());
+}
+```
+The `@When(...)` lines glues this `iExportCoreCne` method to the Gerkin phrase `I export CORE CNE at {string}`, 
+where `{string}` is a Gerkin input argument (that representes the timestamp).
```
Comment (OpenSuze) [inline tests/README.md:57]: represents

File: tests/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+The `@When(...)` lines glues this `iExportCoreCne` method to the Gerkin phrase `I export CORE CNE at {string}`, 
+where `{string}` is a Gerkin input argument (that representes the timestamp).  
+As you can see, the body of the `iExportCoreCne` method simply calls the CORE CNE API of OpenRAO.  
+
+OpenRAO's glue code is located in the [test](src/test/java/com/powsybl/openrao/tests) directory.
+
+## Contributing
+If your development changes a behavior, you may need to update the tests.  
+- If you add a new behavior to OpenRAO, try to add new "scenarios" that test this behavior, in different contexts
+- If you changed or improved some behavior and broke a test, you have to update it. Whenever possible, instead of only
```
Comment (OpenSuze) [inline tests/README.md:65]: you would have to update it

**Review by OpenSuze — APPROVED** [review]

**Review by OpenSuze — APPROVED** [review]

**Review by phiedw — APPROVED** [review]

**Pull request merged**